ush/hpssarch_gen.sh

```diff
@@ -106,12 +106,15 @@ if [ $type = "gfs" ]; then
   fi
   echo  "${dirname}${head}pgrb2.0p25.anl                   " >>gfsa.txt
   echo  "${dirname}${head}pgrb2.0p25.anl.idx               " >>gfsa.txt
-  echo  "${dirname}avno.t${cyc}z.cyclone.trackatcfunix     " >>gfsa.txt
-  echo  "${dirname}avnop.t${cyc}z.cyclone.trackatcfunix    " >>gfsa.txt
-  echo  "${dirname}trak.gfso.atcfunix.${PDY}${cyc}         " >>gfsa.txt
-  echo  "${dirname}trak.gfso.atcfunix.altg.${PDY}${cyc}    " >>gfsa.txt
-  echo  "${dirname}storms.gfso.atcf_gen.${PDY}${cyc}       " >>gfsa.txt
-  echo  "${dirname}storms.gfso.atcf_gen.altg.${PDY}${cyc}  " >>gfsa.txt
+  #Only generated if there are cyclones to track
+  if [ -s ${dirname}avno.t${cyc}z.cyclone.trackatcfunix ]; then
+    echo  "${dirname}avno.t${cyc}z.cyclone.trackatcfunix     " >>gfsa.txt
+    echo  "${dirname}avnop.t${cyc}z.cyclone.trackatcfunix    " >>gfsa.txt
+    echo  "${dirname}trak.gfso.atcfunix.${PDY}${cyc}         " >>gfsa.txt
+    echo  "${dirname}trak.gfso.atcfunix.altg.${PDY}${cyc}    " >>gfsa.txt
+    echo  "${dirname}storms.gfso.atcf_gen.${PDY}${cyc}       " >>gfsa.txt
+    echo  "${dirname}storms.gfso.atcf_gen.altg.${PDY}${cyc}  " >>gfsa.txt
+  fi
```

Comment:
Might I suggest the following:
```suggestion
cyclone_files=(avno.t${cyc}z.cyclone.trackatcfunix
               avnop.t${cyc}z.cyclone.trackatcfunix
               trak.gfso.atcfunix.${PDY}${cyc}
               trak.gfso.atcfunix.altg.${PDY}${cyc}
               storms.gfso.atcf_gen.${PDY}${cyc}
               storms.gfso.atcf_gen.altg.${PDY}${cyc})
for file in ${cyclone_files[@]}; do
  [[ -s ${$dirname}${file} ]] && echo "${dirname}$file" >> gfsa.txt
done
```
This will ensure that each file is checked for existence before being added to be archived.

Reply:
This is a much better solution.  I've implemented and tested it on Hera, which archived two of the six files.  Thanks @aerorahul!